### PR TITLE
[opentitantool] Fix OTP tests to work under bazel

### PR DESCRIPTION
**Pull request by jon-flatley**
Moves OTP test data to a more appropriate location and adds them to the
bazel BUILD file to fix "file not found" errors encountered when running
`bazel run //sw/host/opentitanlib:opentitanlib_test`.

Signed-off-by: Jon Flatley <jflat@google.com>

### CODE REVIEW

**Review by drewmacrae — APPROVED** [review]
LGTM, I was getting ready to reach out and ask if you knew why this test was broken, because I couldn't figure it out. I'm going to check the bazel step when it finishes in CI to see if there's a warning and if it's related, but thanks for working on this test

**Comment by drewmacrae** [comment]
There's an unrelated failure on "bazel build and test" that prevented building and running this test, so there's nothing wrong with it on that side.

**Review by alphan — APPROVED** [review]

**Comment by alphan** [comment]
It looks like CW310 build failed in CI but I don't think it's related to this PR. Any objections to merging this?

File: sw/host/opentitanlib/src/otp/mod.rs [hunk 1]
```diff
@@ -16,15 +16,21 @@ mod tests {
     use anyhow::Result;
     use std::path::Path;
 
+    pub(crate) fn testdata(s: &str) -> String {
+        let mut result = "sw/host/opentitanlib/src/otp/testdata/".to_string();
```
Comment (mcy) [inline sw/host/opentitanlib/src/otp/mod.rs:20]: Would a PathBuf be more appropriate?

Comment (alphan) [reply sw/host/opentitanlib/src/otp/mod.rs:20]: It looks like `sw/host/opentitanlib/src/util/image.rs` also uses a very similar function. Created #10545 to merge this now.

**Pull request merged**